filewatcher/filewatcher_test.go

```diff
@@ -20,6 +21,27 @@ func parser(f io.Reader) (interface{}, error) {
 	return io.ReadAll(f)
 }
 
+func dirParser(f io.Reader) (interface{}, error) {
+	data := []byte{}
+	drc := f.(filewatcher.DummyReadCloser)
+	err := filepath.Walk(drc.Path,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.Mode().IsDir() {
+				return nil
+			}
+			// parse file
+			data, err = ioutil.ReadFile(path)
```

Comment:
so you are overriding `data` with every file walked, in the end you are returning the `data` from the last file walked and throw away all the previous reads. that certainly doesn't sound right to me.

Reply:
also `ioutil` is already deprecated. use `os.ReadFile` instead.

Reply:
Yes that is what Im doing here, but during tests only one file is written so the data turns out to be the same. In the tests I show that the data returned is the same wether you watch the file or the dir that holds it.

Also, I will remove ioutil thanks for keeping me honest